azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+        self.label_uuid = None
+        self.label1_data = []
+        self.label2_data = []
+        self.no_label_data = []
+
+
+class AzConfigurationClientTest(AzureMgmtTestCase):
+    def setUp(self):
+        self.working_folder = os.path.dirname(__file__)
+        super(AzConfigurationClientTest, self).setUp()
+
+        connection_str = os.environ["AZ_CONFIG_CONNECTION"]
+        self.app_config_client = AzureConfigurationClient(connection_str)
+        self.test_data = conftest.setup_data()
+
+    def tearDown(self):
+        conftest.teardown_data(self.test_data)
+
+    # method: add_configuration_setting
+    def test_add_configuration_setting(self):
+
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add_" + self.test_data.key_uuid
+        kv.label = self.test_data.label1
+        kv.value = "test value"
+        kv.content_type = "test content type"
+        kv.tags = {"tag1": "tag1", "tag2": "tag2"}
+        kv.etag = "not needed"  # this etag will not be processed.
+        created_kv = self.app_config_client.add_configuration_setting(kv)
+        self.app_config_client.delete_configuration_setting(kv.key, kv.label)
+        assert (
+            created_kv.label == kv.label
+            and kv.value == kv.value
+            and created_kv.content_type == kv.content_type
+            and created_kv.tags == kv.tags
+        )
+        assert (
+            created_kv.etag is not None
+            and created_kv.last_modified is not None
+            and created_kv.locked is False
+        )
+
+    def test_add_existing_configuration_setting(self):
+        kv = ConfigurationSetting()
+        compare_kv = self.test_data.label1_data[0]
+        kv.key = compare_kv.key
+        kv.label = compare_kv.label
+        kv.value = compare_kv.value
+        kv.content_type = compare_kv.content_type
+
+        with pytest.raises(CloudError):
+            self.app_config_client.add_configuration_setting(kv)
+
+    def test_add_configuration_setting_no_label(self):
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add" + self.test_data.key_uuid
+        # kv.label is None by default
+        kv.value = "test value"
+        kv.content_type = "test content type"
+        kv.tags = {"tag1": "tag1", "tag2": "tag2"}
+        kv.etag = "not needed"  # this etag will not be processed.
+        created_kv = self.app_config_client.add_configuration_setting(kv)
+        self.app_config_client.delete_configuration_setting(kv.key, kv.label)
+        assert (
+            kv.value == kv.value
+            and created_kv.content_type == kv.content_type
+            and created_kv.tags == kv.tags
+        )
+        assert (
+            created_kv.etag is not None
+            and created_kv.last_modified is not None
+            and created_kv.locked is False
+            and created_kv.label is None
+        )
+
+    def test_add_existing_configuration_setting_no_label(self):
```

Comment:
Same question here about the purpose. Are you validating the functionality of the service? or the functionality of the SDK